federatedscope/core/auxiliaries/data_builder.py

```diff
@@ -118,6 +118,238 @@ def _generate_data(client_num=5,
     return data, config
 
 
+def load_external_data(config=None):
```

Comment:
This function appears to do a variety of things (with several internal sub-functions). It would be nice to have a document at the beginning that explains its input and output and the logical work flow.